From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Sun, 29 Mar 2026 23:10:03 +0200
Subject: Allow service's start even if graphical-session is not reached.

While this is not ideal, we need to support desktop environments that do not
bind to graphical-session.target, at least as a temporary measure.

We keep the After=graphical-session.target stanza, since 'After' will only
affect the ordering of start operations if both are queued up to start, while
it is a no-op if graphical-session.target is not pending start.

This partially reverts commit 4d284de29d1d0740c9b80b634631a8f253287680.

Bug: https://github.com/flatpak/xdg-desktop-portal/issues/1983
Bug-Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=2481764
Bug-Ubuntu: https://launchpad.net/bugs/2144855
---
 src/xdg-desktop-portal.service.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/xdg-desktop-portal.service.in b/src/xdg-desktop-portal.service.in
index 696061a..ce09067 100644
--- a/src/xdg-desktop-portal.service.in
+++ b/src/xdg-desktop-portal.service.in
@@ -1,7 +1,8 @@
 [Unit]
 Description=Portal service
 PartOf=graphical-session.target
-Requisite=graphical-session.target
+Requires=dbus.service
+After=dbus.service
 After=graphical-session.target
 
 [Service]
